File: tests/machine/x/clj/dataset_sort_take_limit.clj
(ns main)

(declare products expensive)

(defn -main []
  (def products [{:name "Laptop" :price 1500} {:name "Smartphone" :price 900} {:name "Tablet" :price 600} {:name "Monitor" :price 300} {:name "Keyboard" :price 100} {:name "Mouse" :price 50} {:name "Headphones" :price 200}]) ;; list of map of string to any
  (def expensive (vec (->> (for [p products] p) (sort-by (fn [p] (- (:price p)))) (drop 1) (take 3)))) ;; list of map of string to any
  (println "--- Top products (excluding most expensive) ---")
  (loop [_tmp0 (seq expensive)]
    (when _tmp0
      (let [item (clojure.core/first _tmp0)]
        (let [r (try
          (println (:name item) "costs $" (:price item))
          :next
        (catch clojure.lang.ExceptionInfo e
          (cond
            (= (.getMessage e) "continue") :next
            (= (.getMessage e) "break") :break
            :else (throw e))
          )
        )]
      (cond
        (= r :break) nil
        :else (recur (next _tmp0))
      )
    )
  )
)
)
)

(-main)

File: tests/transpiler/x/php/dataset_sort_take_limit.php
<?php
$products = [["name" => "Laptop", "price" => 1500], ["name" => "Smartphone", "price" => 900], ["name" => "Tablet", "price" => 600], ["name" => "Monitor", "price" => 300], ["name" => "Keyboard", "price" => 100], ["name" => "Mouse", "price" => 50], ["name" => "Headphones", "price" => 200]];
$expensive = [];
foreach ($products as $p) {
  $expensive[] = $p;
}

echo rtrim("--- Top products (excluding most expensive) ---"), PHP_EOL;
foreach ($expensive as $item) {
  echo rtrim((is_float($item["name"]) ? sprintf("%.15f", $item["name"]) : $item["name"]) . " " . "costs $" . " " . (is_float($item["price"]) ? sprintf("%.15f", $item["price"]) : $item["price"])), PHP_EOL;
}
?>
